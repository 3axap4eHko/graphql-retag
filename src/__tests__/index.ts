import { print } from 'graphql/language/printer';
import gql, { resetCaches } from '../index';

beforeEach(() => {
  resetCaches();
});

describe('graphql-reTag test suite', () => {

  test('no fragments test', () => {
    const getAllPosts = gql`
        query GetAllPosts {
            allPosts {
                title
            }
        }
    `;

    expect(print(getAllPosts)).toMatchSnapshot();
  });

  test('single fragment used one time test', () => {
    const image = gql`
        fragment image on Image {
            large
            inline
        }
    `;
    const getAllPosts = gql`
        query GetAllPosts {
            allPosts {
                title
                images {
                    ...${image}
                }
            }
        }
    `;

    expect(print(getAllPosts)).toMatchSnapshot();
  });

  test('a few fragments test', () => {
    const image = gql`
        fragment image on Image {
            large
            inline
        }
    `;
    const comment = gql`
        fragment comment on Comment {
            author {
                username
            }
            content
        }
    `;
    const getAllPosts = gql`
        query GetAllPosts {
            allPosts {
                title
                images {
                    ...${image}
                }
                comments {
                    ...${comment}
                }
            }
        }
    `;

    expect(print(getAllPosts)).toMatchSnapshot();
  });

  test('a few nested fragments used a few times test', () => {
    const image = gql`
        fragment image on Image {
            large
            inline
        }
    `;
    const user = gql`
        fragment user on User {
            username
            avatar {
                ...${image}
            }
        }
    `;
    const getAllPosts = gql`
        query GetAllPosts {
            allPosts {
                title
                author {
                    ...${user}
                }
                images {
                    ...${image}
                }
                comments {
                    author {
                        ...${user}
                    }
                    image {
                        ...${image}
                    }
                    message
                }
            }
        }
    `;

    expect(print(getAllPosts)).toMatchSnapshot();
  });
});