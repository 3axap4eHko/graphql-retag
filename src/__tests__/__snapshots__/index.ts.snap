// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql-reTag test suite a few fragments test 1`] = `
"fragment image on Image {
  large
  inline
}

fragment comment on Comment {
  author {
    username
  }
  content
}

query GetAllPosts {
  allPosts {
    title
    images {
      ...image
    }
    comments {
      ...comment
    }
  }
}
"
`;

exports[`graphql-reTag test suite a few nested fragments test 1`] = `
"fragment user on User {
  username
  avatar {
    ...image
  }
}

fragment image on Image {
  large
  inline
}

fragment comment on Comment {
  author {
    ...user
  }
  content
}

query GetAllPosts {
  allPosts {
    title
    author {
      ...user
    }
    images {
      ...image
    }
    comments {
      ...comment
    }
  }
}
"
`;

exports[`graphql-reTag test suite no fragments test 1`] = `
"query GetAllPosts {
  allPosts {
    title
  }
}
"
`;

exports[`graphql-reTag test suite single fragment test 1`] = `
"fragment image on Image {
  large
  inline
}

query GetAllPosts {
  allPosts {
    title
    images {
      ...image
    }
  }
}
"
`;
